sources/packages/backend/apps/workers/src/controllers/assessment/assessment.controller.ts

```diff
@@ -482,6 +495,65 @@ export class AssessmentController {
         ? output[outputName] + award.total
         : award.total;
     });
+    ftProgramYearContributionTotal?.forEach((ftContributionTotal) => {
+      const outputName = `programYearTotal${ftContributionTotal.contribution}`;
+      output[outputName] = output[outputName]
+        ? output[outputName] + ftContributionTotal.total
+        : ftContributionTotal.total;
+    });
+  }
+
+  /**
+   * Get the promises of the program year awards totals for part time and
+   * full time and contribution totals for full time for the assessment.
+   * @param assessmentId assessment id.
+   * @param options method options.
+   * - `offeringIntensity` the offering intensity to be used.
+   * - `alternativeReferenceDate` the reference date to be used.
+   * @returns the promise to get the program year totals.
+   */
+  private async getProgramYearTotals(
```

Comment:
This method seems to be a better candidate to be moved to service instead of keeping it in the controller.
Can it be moved to `StudentAssessmentService` maybe?

Reply:
As discussed moved it to AssessmentSequentialProcessingService, as it would not affect the existing abstraction and consumer usage of award and contribution totals.